### ISSUE TITLE
Spectate is defective

### ISSUE THREAD

**Issue by oldmud0** [oldmud0/LilithPort#3]
It just doesn't work. Not sure if it works in the original.
http://www.fmhq.us/2d-fighter-maker-2nd/lilithport-for-english/msg14714/#msg14714

**Comment by Bradsama** [comment]
Users with the Online Community for Terrordrome have confirmed that this doesn't work. In the old version of MTSP and Telepone this feature works. It's possible to find the Last mtsp source and see if you can revert the code back. My community would be willing to test this.

~ AcidSama - FMHQ Owner,Terrordrome Co-Designer

**Comment by oldmud0** [comment]
All right then, it has been confirmed. I will place my efforts on pinpointing the issue.

**Comment by oldmud0** [comment]
I need someone else to do this. FM won't let me have two of the same game open, even if they have different .kgt file names.

**Comment by Bradsama** [comment]
If you want you can send me whatever you got and I can get this tested we
have plenty of people that play Terrordrome online each day.

**Comment by oldmud0** [comment]
That's fine, I just need one person to start the game while I am spectating them so that I can generate a nice pretty exception while I'm debugging. But while you're at it, maybe you can have the pre-release tested?

EDIT: I got a few Vanguard Princess players on and it seems that they can spectate just fine. Works for Type Wild fine too. Is 2DFM or LilithPort crashing? All I need are a few error logs.

**Comment by Bradsama** [comment]
@oldmud0 2DFM is crashing when me and my other dev try to spectate. But at one point in time it worked but I got a error after awhile saying "ERROR: Could not synchronize; leaving spectator mode...
"
My connection isn't the best so he is going to test it with someone else that has better internet than I do.

**Comment by Bradsama** [comment]
After more testing it started working for me and was stable. But whenever I
was getting spectated it was reported to him that nothing happened on my
watchers end the game didn't even load.

**Comment by oldmud0** [comment]
It sounds like LilithPort is having trouble receiving packets. I'm not sure why that would happen though; the latency would have to be through the roof for things not to happen in sync.

From my understanding, game frame data is sent through the server and to the spectators if you are a client. If you are a host, then it relies on your ports being opened for spectators to connect directly to you.

**Comment by Bradsama** [comment]
Ooh ok will it still has some issues some are able to spectate while others are not and their game crashes.

Btw chat sound isn't working at all with this Pre-Release

**Comment by oldmud0** [comment]
??? Really? It's supposed to mute while in game, but I'll check it.

**Comment by Bradsama** [comment]
Yea if ya need a way to test connect to terrordrome server at
trdm.duckdns.org port 7500

People on there just about all the time

**Comment by Bradsama** [comment]
Got more reports looks like spectate works for some but the FM2nd will
crash sometimes..

Also Fm2nd was crashing when someone was trying to send a challenge to
someone very strange things.

**Comment by Bradsama** [comment]
Any news on when this could be fixed we are holding out on releasing a new version of Terrordrome till your able to get spectate working 100%

**Comment by oldmud0** [comment]
No, I don't know how to fix this yet, and school is killing me.. but I'm glad you said that, because now fixing this bug will be top priority.

**Comment by Dzierzan** [comment]
Hello. I understand you're very busy with school and stuff. I'll say this way. Spectating mode does work and it doesn't. Just sometimes it works flawlessly, but with some people it doesn't. I even think it might be not fully fixable. Anyway, can you at least try to fix not working sounds and blinking when someone writes something? It appears only "announcement" sound works.

Thank in advance!

**Comment by oldmud0** [comment]
The sounds and blinking have been fixed.
With the spectate, I'm suspecting that it's either a problem with the server (?) or because one of the two people needs to be an actual host in order to directly spectate a match.

**Comment by Dzierzan** [comment]
I think I tried every combination. Either it was random or only one side could spectate the second player. The 2nd one couldn't spectate 1st one. I think I tested it with at least 3 people. I had similar result. Hey, at least it's somehow working! So I appreciate your work.

Regards

**Comment by oldmud0** [comment]
Well, if you ever encounter an exception or something I can look at, please post it here. Be sure to wrap your exception with code (```) tags.

**Comment by Dzierzan** [comment]
Do you mean anything apart from spectating mode?

"Be sure to wrap your exception with code (```) tags." Can you write an example? To be honest, I only made an account here to post something here.

Well, is it possible to add message that the game is desynchronized? I know Telepone has that feature. Ohh and I noticed that in order to play with someone, you have to relogin. Command /reload doesn't really help.

**Comment by oldmud0** [comment]
These are really weird problems you're having. Does the same stuff happen when you try making a server yourself?
Example of code tags:
```
Hello
```
turns into

```
Hello
```

**Comment by Dzierzan** [comment]
Basically everyone in Bradsama's server have this issue. Relogin is not really a problem, but sometimes it's annoying or you have to tell newbies how it works.

Do you know perhaps other games which use 2nd Fighther Maker engine which also use lilithport translated by you?

**Comment by oldmud0** [comment]
There is just Type Wild which was translated by me. That's about it.

**Comment by Bradsama** [comment]
Any Fighter Maker 2nd game will work with Lilithport. It's still the same
core coding from the original MTSP.

**Comment by oldmud0** [comment]
Oh, I thought you were asking what games recommend using LilithPort..
